corehq/apps/users/models.py

```diff
@@ -800,6 +813,8 @@ class CouchUser(Document, DjangoUserMixin, IsMemberOfMixin, UnicodeMixIn, EulaMi
     assigned_location_ids = StringListProperty()
     has_built_app = BooleanProperty(default=False)
 
+    reporting_metadata = SchemaProperty(ReportingMetadata)
```

Comment:
+1 to having it separated from the user model. I also think it should be tracked per device / user combo and combined with [DeviceIdLastUsed](https://github.com/dimagi/commcare-hq/blob/7bd298e2d8870bca8ad32ab77b61c3ffac94fb27/corehq/apps/users/models.py#L766-L766)